src/middleware/src/http.ts

```diff
@@ -0,0 +1,56 @@
+import { Middleware } from 'kinvey-js-sdk/dist/request';
+import { request as HttpRequest } from 'http';
+import { device } from 'platform'
+
+function deviceInformation() {
+  const platform = device.os;
+  const version = device.osVersion;
+  const manufacturer = device.manufacturer;
+
+  // Return the device information string.
+  // const parts = [`js-${pkg.name}/${pkg.version}`];
+  const parts = ['js-kinvey-nativescript-sdk/3.5.0'];
+
+  return parts.concat([platform, version, manufacturer]).map((part) => {
+    if (part) {
+      return part.toString().replace(/\s/g, '_').toLowerCase();
+    }
+
+    return 'unknown';
+  }).join(' ');
+}
+
+export class HttpMiddleware extends Middleware {
+  constructor(name = 'Http Middleware') {
+    super(name);
+  }
+
+  handle(request: any) {
+    const { url, method, headers, body, timeout, followRedirect } = request;
+
+    // Add the X-Kinvey-Device-Information header
+    headers['X-Kinvey-Device-Information'] = deviceInformation();
+
+    return HttpRequest({
```

Comment:
@thomasconner nothing wrong with this, but I'm curious why xhr didn't work. Can you try to find out? Afaik, xhr should work just fine.